src/components/AuthForm.js

```diff
@@ -1,147 +1,123 @@
 import React, { useState } from "react";
-import Form from "react-bootstrap/Form";
-import FormField from "./FormField";
-import Button from "react-bootstrap/Button";
+import { Row, Col, Form, Button, Container } from "react-bootstrap";
 import Spinner from "react-bootstrap/Spinner";
 import { useAuth } from "./../util/auth.js";
-import { useForm } from "react-hook-form";
 
 function AuthForm(props) {
+  const { onAuth } = props;
   const auth = useAuth();
 
   const [pending, setPending] = useState(false);
-  const { handleSubmit, register, errors, getValues } = useForm();
-
-  const submitHandlersByType = {
-    signin: ({ email, pass }) => {
-      return auth.signin(email, pass).then((user) => {
-        // Call auth complete handler
-        props.onAuth(user);
-      });
-    },
-    signup: ({ email, pass }) => {
-      return auth.signup(email, pass).then((user) => {
-        // Call auth complete handler
-        props.onAuth(user);
-      });
-    },
-    forgotpass: ({ email }) => {
-      return auth.sendPasswordResetEmail(email).then(() => {
-        setPending(false);
-        // Show success alert message
-        props.onFormAlert({
-          type: "success",
-          message: "Password reset email sent",
-        });
-      });
-    },
-    changepass: ({ pass }) => {
-      return auth.confirmPasswordReset(pass).then(() => {
-        setPending(false);
-        // Show success alert message
-        props.onFormAlert({
-          type: "success",
-          message: "Your password has been changed",
-        });
-      });
-    },
+  const [showOTP, setShowOTP] = useState(false);
+  const [state, setState] = useState({
+    phoneNumber: "",
+    otp: "",
+  })
+  
+  const onChangeHandler = (event) => {
+    const { name, value } = event.target;
+    setState({...state, [name]: value });
   };
 
-  // Handle form submission
-  const onSubmit = ({ email, pass }) => {
-    // Show pending indicator
+  const requestOTPCode = async (e) => {
+    e.preventDefault();
     setPending(true);
+    auth.setUpRecaptcha();
+    const otpRequestSent = await auth.requestOTPCode(state.phoneNumber);
+    setPending(false);
+    if (otpRequestSent) {
+      setShowOTP(true);
+    } 
+  }
 
-    // Call submit handler for auth type
-    submitHandlersByType[props.type]({
-      email,
-      pass,
-    }).catch((error) => {
-      setPending(false);
-      // Show error alert message
-      props.onFormAlert({
-        type: "error",
-        message: error.message,
-      });
-    });
+  const onSubmitOtp = async (e) => {
+    e.preventDefault();
+    setPending(true);
+    let otpInput = state.otp;
+    const userData = {
```

Comment:
I'm guessing these were test values, and should be set to the actual data from the event?

Reply:
No longer here